File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int j=0, i=0;
        while (i<nums.size()){
            if (j==0 || j==1 || nums[j-2]!=nums[i]){
                nums[j] = nums[i];
                j+=1;
            }
            i+=1;
        }

        return j;
    }
};

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.go
func removeDuplicates(nums []int) int {
    cnt, k, i := 1, 2, 1
    for j := 1; j < len(nums); j++ {
        if nums[j] == nums[j-1] {
            if cnt < k {
                nums[i] = nums[j]
                i++
            }
            cnt++
        } else {
            nums[i] = nums[j]
            cnt = 1
            i++
        }
    }

    return i
}